File: app/src/main/java/com/innovation4you/napking/model/OccupancyEntry.java
package com.innovation4you.napking.model;

import android.support.annotation.NonNull;

import java.util.Locale;

public class OccupancyEntry implements Comparable<OccupancyEntry> {

	public int weekDay;

	public int hour;

	public int minute;

	public int occupancy;

	public OccupancyEntry(int weekDay, int hour, int occupancy) {
		this.weekDay = weekDay;
		this.hour = hour;
		this.occupancy = occupancy;
	}

	@Override
	public int compareTo(@NonNull OccupancyEntry another) {
		int compare = compareInt(weekDay, another.weekDay);
		if (compare == 0) {
			compare = compareInt(hour, another.hour);
			if (compare == 0) {
				compare = compareInt(minute, another.minute);
			}
		}
		return compare;
	}

	private static int compareInt(int lhs, int rhs) {
		return lhs < rhs ? -1 : (lhs == rhs ? 0 : 1);
	}

	@Override
	public String toString() {
		return String.format(Locale.ENGLISH, "%d-%d: %d", weekDay, hour, occupancy);
	}
}

File: app/src/main/java/com/innovation4you/napking/model/OccupancyEntry.kt
package com.innovation4you.napking.model

import java.util.*

class OccupancyEntry(var weekDay: Int, var hour: Int, var occupancy: Int) : Comparable<OccupancyEntry> {

    var minute: Int = 0

    override fun compareTo(another: OccupancyEntry): Int {
        var compare = compareInt(weekDay, another.weekDay)
        if (compare == 0) {
            compare = compareInt(hour, another.hour)
            if (compare == 0) {
                compare = compareInt(minute, another.minute)
            }
        }
        return compare
    }

    private fun compareInt(lhs: Int, rhs: Int): Int {
        return if (lhs < rhs) -1 else if (lhs == rhs) 0 else 1
    }

    override fun toString(): String {
        return String.format(Locale.ENGLISH, "%d-%d: %d", weekDay, hour, occupancy)
    }
}
